xmodule/library_content_block.py

```diff
@@ -451,7 +451,12 @@ def author_view(self, context):
         context['is_loading'] = is_updating
 
         # The following JS is used to make the "Update now" button work on the unit page and the container view:
-        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/library_content_edit.js'))
+        if 'library' in root_xblock.category:
```

Comment:
won't this always be the case? If so, should we instead say just

if root_xblock.source_library_id and len(root_xblock.children) > 0:
    fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/library_content_edit.js'))
    
 The duplicate fragment.add code confused me for a moment, so looking to simplify.

Reply:
For some reason this code runs on the unit page so an error is thrown `VerticalXblockMixins` not having the attribute `source_library_id`. To prevent the error we need to check that the root block is a library block which fall under two types `library` and `library_content`